homeassistant/components/squeezebox/media_player.py

```diff
@@ -451,6 +457,66 @@ async def async_play_media(
             )
             media_id = play_item.url
 
+        if announce:
```

Comment:
```suggestion
        if announce:
          cmd = "announce"
```
How about this approach? "announce" as a cmd. That way we can use all of the existing logic to load playlists and urls. 

https://github.com/rajlaud/pysqueezebox/tree/play-announcement

Reply:
yes, that would work - and then move all of the logic into the library

Reply:
I can make that change, but I can't test anything ATM as I'm away.  Maybe let me know the changes you'd suggest overall and I'll push a new version when you're ready.  I'm not back for a couple of weeks to test it.

Reply:
All done, pending a new library PR